EventFilter/HcalRawToDigi/interface/HcalDataFrameFilter.h

```diff
@@ -28,14 +29,21 @@ class HcalDataFrameFilter {
   HcalCalibDigiCollection filter(const HcalCalibDigiCollection& incol, HcalUnpackerReport& r);
   /// filter ZDC data frames
   ZDCDigiCollection filter(const ZDCDigiCollection& incol, HcalUnpackerReport& r);
+  /// filter QIE10 data frames
+  QIE10DigiCollection filter(const QIE10DigiCollection& incol, HcalUnpackerReport& r);
+  /// filter QIE11 data frames
+  QIE11DigiCollection filter(const QIE11DigiCollection& incol, HcalUnpackerReport& r);
   /// whether any filters are on
   bool active() const;
+  /// get conditions
+  void setConditions(const HcalDbService* conditions);
 private:
   bool requireCapid_;
   bool requireDVER_;
   bool energyFilter_;
   int firstSample_, lastSample_;
   double minimumAmplitude_;
+  const HcalDbService* conditions_;
```

Comment:
Only the HcalRawToDigi class uses the HcalDataFrameFilter:

https://github.com/cms-sw/cmssw/search?utf8=%E2%9C%93&q=HcalDataFrameFilter&type=

and HcalRawToDigi initializes the energy filter to false:

https://github.com/cms-sw/cmssw/blob/09c3fce6626f70fd04223e7dacebf0b485f73f54/EventFilter/HcalRawToDigi/plugins/HcalRawToDigi.cc#L17

so I think that the energy filter code is unreachable in any case -- also a strange design.